src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinNamesAnnotationIntrospector.kt

```diff
@@ -25,12 +20,24 @@ import kotlin.reflect.jvm.javaType
 import kotlin.reflect.jvm.kotlinFunction
 
 internal class KotlinNamesAnnotationIntrospector(val module: KotlinModule, val cache: ReflectionCache, val ignoredClassesForImplyingJsonCreator: Set<KClass<*>>) : NopAnnotationIntrospector() {
-    /*
-    override public fun findNameForDeserialization(annotated: Annotated?): PropertyName? {
-        // This should not do introspection here, only for explicit naming by annotations
+
+    /**
+     * resolve the name of the property, if not given otherwise.
+     */
+    override fun findNameForSerialization(annotated: Annotated?): PropertyName? {
+        if (annotated is AnnotatedMethod) {
```

Comment:
Might want to add:

a.declaringClass.methods.any { it.name == "box-impl" }

to ensure that this is an inline class. Or declaredMethods. I always forget which is which.

Reply:
Hmm, I don't see any methods with `box-impl`.  The [documentation on name mangling](https://github.com/Kotlin/KEEP/blob/master/proposals/inline-classes.md) doesn't mention that name, but [an earlier KEEP](https://github.com/Kotlin/KEEP/blob/master/proposals/inline-classes.md#inline-classes-abi-jvm) does.  Do we need another unit test to show this behavior?